beacon-chain/powchain/service.go

```diff
@@ -962,13 +958,19 @@ func (s *Service) validateDepositContainers(ctrs []*protodb.DepositContainer) bo
 	if ctrLen == 0 {
 		return true
 	}
+	// Sort deposits in ascending order.
+	sort.Slice(ctrs, func(i, j int) bool {
+		return ctrs[i].Index < ctrs[j].Index
+	})
+	startIndex := int64(0)
 	for _, c := range ctrs {
-		if c.Index == 0 {
-			return true
+		if c.Index != startIndex {
+			log.Info("Recovering missing deposit containers, node is re-requesting missing deposit data")
```

Comment:
I think this log should be placed outside of `validateDepositContainers` because the function itself does not know what will happen when it returns `false`. It should be the caller that logs out what it's going to do.

Reply:
This was discussed before too and is a valid point, but we initially decided against having the caller log it. If you look at how this method is called, it basically is in the last condition of the if statement, if we want to specifically log it then we will have to basically have a more complex nested if checks for the first 2 conditions and then finally this method. If you feel strongly on it , I can amend the caller so that its logged by the caller.